include/index_helper.cuh

```diff
@@ -336,6 +336,29 @@ namespace quda {
     return 2 * cb_index + x1odd;
   }
 
+  /** Compute the 1-d checkerboard index and parity from
+      the full linear lattice (not what's used for
+      indexing into fields, re:padding argument for getIndexFull).
+
+      @param[out] out_parity Output site parity
+      @param[in] X full lattice dimensions
+      @param[in] full_index full linear lattice index
+      @return 1-d checkerboard index
+   */
+  template <typename I>
+  __device__ __host__ inline int getParityCBFromFull(int& out_parity, const I X[4], const int full_index) {
```

Comment:
Just a comment: I believe there are more index utility functions in this file than necessary, it might take some cleaning in the future.